ftp-test-utils/src/main/java/org/ballerinalang/stdlib/ftp/testutils/mockServerUtils/TwoFactorAuthorizedKeysAuthenticator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.ftp.testutils.mockServerUtils;
+
+import org.apache.sshd.server.config.keys.AuthorizedKeysAuthenticator;
+import org.apache.sshd.server.session.ServerSession;
+
+import java.io.File;
+import java.security.PublicKey;
+
+public class TwoFactorAuthorizedKeysAuthenticator extends AuthorizedKeysAuthenticator {
+
+    private static SftpAuthStatusHolder sftpAuthStatusHolder;
+
+    public TwoFactorAuthorizedKeysAuthenticator(File file, SftpAuthStatusHolder authStatusHolder) {
+        super(file);
+        sftpAuthStatusHolder = authStatusHolder;
+    }
+
+    @Override
+    public boolean authenticate(String username, PublicKey key, ServerSession session) {
```

Comment:
This method always return `false`. Right?

Reply:
Yes. I did this to enable tow factor authentication. Because otherwise it would skip the password based authentication, as the framework is designed so that only a single type of authentication is enough.